src/IROperator.h

```diff
@@ -900,12 +901,50 @@ inline Expr select(Expr condition, Expr true_value, Expr false_value) {
  * which can accept multiple conditions and values in pairs. Evaluates
  * to the first value for which the condition is true. Returns the
  * final value if all conditions are false. */
-template<typename... Args,
-         typename std::enable_if<Halide::Internal::all_are_convertible<Expr, Args...>::value>::type* = nullptr>
+template<typename... Args>
 inline Expr select(Expr c0, Expr v0, Expr c1, Expr v1, Args&&... args) {
     return select(std::move(c0), std::move(v0), select(std::move(c1), std::move(v1), std::forward<Args>(args)...));
 }
 
+/** Equivalent of ternary select(), but taking/returning tuples. If the condition is
+ * a Tuple, it must match the size of the true and false Tuples. */
+// @{
+inline Tuple tuple_select(const Tuple &condition, const Tuple &true_value, const Tuple &false_value) {
+    user_assert(condition.size() == true_value.size() && true_value.size() == false_value.size())
```

Comment:
Anything that throws a user error with inline linkage needs to be marked as NO_INLINE. If it gets a stack frame, then introspection can report an error on the right line number in user code. If it doesn't get a stack frame you don't end up with a useful error message.

The alternative is to move the implementation into IROperator.cpp

Reply:
OK, but the version of `select()` just above this violates that too. (I'll move both of them into the cpp file.)